packages/core/src/selectable/big-number-field.ts

```diff
@@ -73,16 +74,34 @@ export class BigNumberField<EntityT extends Entity> extends BigNumberFieldBase<E
  * @typeparam EntityT Type of the entity the field belongs to
  */
 export class ComplexTypeBigNumberPropertyField<EntityT extends Entity> extends BigNumberFieldBase<EntityT> {
+  /**
+   * The constructor of the entity or the complex type this field belongs to
+   */
+  readonly fieldOf: ConstructorOrField<EntityT>;
+
   /**
+   * Creates an instance of ComplexTypeBigNumberPropertyField.
+   *
+   * @param fieldName Actual name of the field used in the OData request
+   * @param fieldOf The constructor of the entity or the complex type this field belongs to
+   * @param edmType Type of the field according to the metadata description
+   */
+  constructor(fieldName: string, fieldOf: ConstructorOrField<EntityT>, edmType: EdmType);
+  /**
+   * @deprecated since verision 1.19.0
+   *
    * Creates an instance of ComplexTypeBigNumberPropertyField.
    *
    * @param fieldName Actual name of the field used in the OData request
    * @param entityConstructor Constructor type of the entity the field belongs to
    * @param parentTypeName Name of the parent complex type
    * @param edmType Type of the field according to the metadata description
    */
-  constructor(fieldName: string, entityConstructor: Constructable<EntityT>, readonly parentTypeName: string, edmType: EdmType) {
-    super(fieldName, entityConstructor, edmType);
+  constructor(fieldName: string, entityConstructor: Constructable<EntityT>, parentTypeName: string, edmType: EdmType);
+
+  constructor(fieldName: string, fieldOf: ConstructorOrField<EntityT>, arg3: string | EdmType, arg4?: EdmType) {
```

Comment:
Should we also deprecate this (although it is new) and add documentation? Or maybe just a short documentation, that this is the union of the two constructors before?

Reply:
ok I add a little sentence